File: src/main/java/com/github/derdan/explorerextension/SRecordFileType.java
package com.github.derdan.explorerextension;

import com.intellij.openapi.fileTypes.LanguageFileType;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class SRecordFileType extends LanguageFileType {
    public static final SRecordFileType INSTANCE = new SRecordFileType();

    private SRecordFileType() {
        super(SRecordLanguage.INSTANCE);
    }

    @NotNull
    @Override
    public String getName() {
        return "SRecord File";
    }

    @NotNull
    @Override
    public String getDescription() {
        return "SRecord text file containing binary data";
    }

    @NotNull
    @Override
    public String getDefaultExtension() {
        return "srec";
    }

    @Nullable
    @Override
    public Icon getIcon() {
        return SRecordIcon.FILE;
    }

}

File: src/main/kotlin/com/github/derdan/explorerextension/SRecordFileType.kt
package com.github.derdan.explorerextension

import com.intellij.openapi.fileTypes.LanguageFileType
import javax.swing.Icon

class SRecordFileType private constructor() : LanguageFileType(SRecordLanguage.INSTANCE) {
    override fun getName(): String {
        return "SRecord File"
    }

    override fun getDescription(): String {
        return "SRecord text file containing binary data"
    }

    override fun getDefaultExtension(): String {
        return "srec"
    }

    override fun getIcon(): Icon? {
        return SRecordIcon.FILE
    }

    companion object {
        @JvmField
        val INSTANCE = SRecordFileType()
    }
}
